ResearchKit/Common/ORKAnswerFormat.h

```diff
@@ -109,12 +111,14 @@ ORK_CLASS_AVAILABLE
                                                                    defaultValue:(double)defaultValue
                                                           maximumFractionDigits:(NSInteger)maximumFractionDigits
                                                                        vertical:(BOOL)vertical
+                                                              hideSelectedValue:(BOOL)shouldHideValue
```

Comment:
Hey @bgannin , do you think it is necessary to add this parameter to the designated initializer? My concern is that this may not be backwards compatible..

Can't this simply remain as a bool property that defaults to NO? Would love to hear your thoughts about this..

Reply:
Hi @srinathtm-apple, I went back and forth on this one - for completeness of configuration in the default initializer it seems to fit there. That said, a convenience initializer to match the prior signature would ease the transition and ensure backward compatibility. I'll add one 👍

Reply:
To add a bit more…

I agree that these class methods can be reverted to not include the value for compatibility and stick with this change as a property essentially. That said, it makes sense to me to have the designated initializer expand to include with a convenience equivalent to the prior one (in the same way that min/max descriptions are optional but included for completeness of setup.)

Reply:
@bgannin thanks for the consideration! This might be part of a bigger discussion, but we need to establish some trade-offs between completeness of configuration and overhead of adding convenience initializers.

For this particular use case, I can totally see developers wanting to leverage this in their implementation immediately by adding a new line which sets the property hideSelectedValue to YES. And for folks implementing a scale answer format for the very first time, does adding a fourth initializer option help them or will it just end up cluttering Xcode's auto complete scroll window?

Would appreciate your thoughts/feedback with regards to this.

Reply:
@srinathtm-apple That's a good point and I'll defer to the larger discussion for the framework. Will roll back the initializer changes and do it as an editable property later today.